enterprise/kernel/src/main/java/org/neo4j/kernel/enterprise/builtinprocs/EnterpriseBuiltInDbmsProcedures.java

```diff
@@ -527,4 +524,14 @@ public QueryTerminationResult( QueryId queryId, String username )
             this.connectionCount = connectionCount;
         }
     }
+
+    public static class MetadataResult
+    {
+        public final Map<String,Object> metadata;
```

Comment:
The original idea is that people will call this procedure and will get a map as a result so they can easily find a value that they set in metadata by doing a lookup, having results of this procedure as a list of key-value pairs will make it more complex and cumbersome to do from a user perspective.